spring-kafka/src/main/java/org/springframework/kafka/aot/KafkaRuntimeHintsRegistrar.java

```diff
@@ -0,0 +1,253 @@
+import org.springframework.kafka.support.serializer.DelegatingSerializer;
+import org.springframework.kafka.support.serializer.ErrorHandlingDeserializer;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+import org.springframework.kafka.support.serializer.JsonSerializer;
+import org.springframework.kafka.support.serializer.ParseStringDeserializer;
+import org.springframework.kafka.support.serializer.StringOrBytesSerializer;
+import org.springframework.kafka.support.serializer.ToStringSerializer;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.handler.annotation.MessageMapping;
+import org.springframework.util.ClassUtils;
+
+/**
+ * {@link RuntimeHintsRegistrar} for Spring for Apache Kafka.
+ *
+ * @author Gary Russell
+ * @since 3.0
+ *
+ */
+public class KafkaRuntimeHintsRegistrar implements RuntimeHintsRegistrar {
+
+	@Override
+	public void registerHints(RuntimeHints hints, @Nullable ClassLoader classLoader) {
+		ReflectionHints reflectionHints = hints.reflection();
+		Stream.of(
+					PartitionOffset.class,
+					TopicPartition.class,
+					ConsumerProperties.class,
+					ContainerProperties.class,
+					ProducerListener.class,
+					KafkaListener.class,
+					EnableKafka.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_METHODS)));
+
+		Stream.of(
+					Message.class,
+					ImplicitLinkedHashCollection.Element.class,
+					KafkaListener.class,
+					MessageMapping.class,
+					KafkaListeners.class,
+					NewTopic.class,
+					AbstractKafkaListenerContainerFactory.class,
+					ConcurrentKafkaListenerContainerFactory.class,
+					KafkaListenerContainerFactory.class,
+					KafkaListenerEndpointRegistry.class,
+					DefaultKafkaConsumerFactory.class,
+					DefaultKafkaProducerFactory.class,
+					KafkaAdmin.class,
+					KafkaOperations.class,
+					KafkaResourceFactory.class,
+					KafkaTemplate.class,
+					ProducerFactory.class,
+					KafkaOperations.class,
+					ConsumerFactory.class,
+					LoggingProducerListener.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_CONSTRUCTORS,
+								MemberCategory.INVOKE_DECLARED_METHODS,
+								MemberCategory.INTROSPECT_PUBLIC_METHODS)));
+
+		Stream.of(
+					Message.class,
+					ImplicitLinkedHashCollection.Element.class,
+					KafkaListener.class,
+					MessageMapping.class,
+					KafkaListeners.class,
+					KafkaListenerAnnotationBeanPostProcessor.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_CONSTRUCTORS,
+								MemberCategory.INVOKE_DECLARED_METHODS,
+								MemberCategory.INTROSPECT_PUBLIC_METHODS)));
+
+		Stream.of(
+					KafkaBootstrapConfiguration.class,
+					CreatableTopic.class,
+					KafkaListenerEndpointRegistry.class)
```

Comment:
I wonder if you tried to go baby steps with all of these types registration.
Looks like you just ported whatever there was in Spring Native...

Reply:
That is exactly correct; I would rather get it working quickly and then we can prune it later.